src/sysusers/sysusers.c

```diff
@@ -1785,27 +1861,33 @@ int main(int argc, char *argv[]) {
 
         if (optind < argc) {
                 int j;
-
-                for (j = optind; j < argc; j++) {
-                        k = read_config_file(argv[j], false);
-                        if (k < 0 && r == 0)
-                                r = k;
+                bool found_override = false;
+
+                if (arg_config_name) {
+                        /* If overrides should be allowed, look for a config file that
+                         * would override our configuration: treat the configuration
+                         * given on the command as if it was in priority between /run
+                         * and /usr/. */
+                        r = read_config_files(override_file_dirs, arg_config_name);
+                        if (r < 0)
+                                goto finish;
+                        found_override = r;
                 }
-        } else {
-                _cleanup_strv_free_ char **files = NULL;
-                char **f;
 
-                r = conf_files_list_nulstr(&files, ".conf", arg_root, 0, conf_file_dirs);
-                if (r < 0) {
-                        log_error_errno(r, "Failed to enumerate sysusers.d files: %m");
+                if (!found_override)
+                        for (j = optind; j < argc; j++) {
```

Comment:
It may be not necessary now, but I prefer to initialize `r = 0` before the for loop. In the future some operation may be inserted just before this.

Reply:
We always exit when `r < 0` so this should be OK.

Reply:
I mean that it might be `r > 0`. Then, `r == 0` in the loop is always negative, and r never updated.